pkgs/by-name/fi/fish-lsp/package.nix

```diff
@@ -0,0 +1,77 @@
+{
+  fetchFromGitHub,
+  fetchYarnDeps,
+  fish,
+  installShellFiles,
+  lib,
+  makeWrapper,
+  nix-update-script,
+  nodejs,
+  npmHooks,
+  stdenv,
+  which,
+  yarnBuildHook,
+  yarnConfigHook,
+}:
+stdenv.mkDerivation rec {
+  pname = "fish-lsp";
+  version = "1.0.8-1";
+
+  src = fetchFromGitHub {
+    owner = "ndonfris";
+    repo = "fish-lsp";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-u125EZXQEouVbmJuoW3KNDNqLB5cS/TzblXraClcw6Q=";
+  };
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
+    hash = "sha256-hHw7DbeqaCapqx4dK5Y3sPut94ist9JOU8g9dd6gBdo=";
+  };
+
+  nativeBuildInputs = [
+    yarnBuildHook
+    yarnConfigHook
+    npmHooks.npmInstallHook
+    nodejs
+    installShellFiles
+    makeWrapper
+    # fish-lsp dependencies
+    fish
+    which
```

Comment:
are these needed at build time or at runtime?

Reply:
Hi @Scrumplex,

Thank you for having look at the PR.

- `fish` is required for the `sh:build-wasm` script which runs https://github.com/ndonfris/fish-lsp/blob/master/scripts/build-fish-wasm.fish 
Error without `fish`
```
> Running phase: buildPhase
       > Executing yarnBuildHook
       > yarn run v1.22.22
       > $ run-s sh:build-time sh:build-wasm
       > warning You don't appear to have an internet connection. Try the --offline flag to use the cache for registry queries.
       > $ fish ./scripts/build-time.fish
       > /bin/sh: fish: not found
       > error Command failed with exit code 127.
       > info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
       > ERROR: "sh:build-time" exited with 127.
       > error Command failed with exit code 1.
```
- `which` is required to generate the completion. See https://github.com/ndonfris/fish-lsp/blob/master/src/utils/builtins.ts (`findShell` spawns `which fish`)
Error without `which`
```
       > Running phase: installPhase
       > /bin/sh: which: not found
       > /nix/store/aam03bwd19cadd1ad6qw26gr7h7hkh7m-fish-lsp-unstable-2024-07-26/share/fish-lsp/out/utils/builtins.js:88
       >     return result.stdout.toString().split('\n');
       >                          ^
       >
       > TypeError: Cannot read properties of null (reading 'toString')
       >     at createFunctionNamesList (/nix/store/aam03bwd19cadd1ad6qw26gr7h7hkh7m-fish-lsp-unstable-2024-07-26/share/fish-lsp/out/utils/builtins.js:88:26)
       >     at Object.<anonymous> (/nix/store/aam03bwd19cadd1ad6qw26gr7h7hkh7m-fish-lsp-unstable-2024-07-26/share/fish-lsp/out/utils/builtins.js:90:29)
       >     at Module._compile (node:internal/modules/cjs/loader:1469:14)
       >     at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
       >     at Module.load (node:internal/modules/cjs/loader:1288:32)
       >     at Module._load (node:internal/modules/cjs/loader:1104:12)
       >     at Module.require (node:internal/modules/cjs/loader:1311:19)
       >     at require (node:internal/modules/helpers:179:18)
       >     at Object.<anonymous> (/nix/store/aam03bwd19cadd1ad6qw26gr7h7hkh7m-fish-lsp-unstable-2024-07-26/share/fish-lsp/out/utils/translation.js:38:20)
       >     at Module._compile (node:internal/modules/cjs/loader:1469:14)
       >
       > Node.js v20.17.0
       > installShellCompletion: installed shell completion file `/nix/store/aam03bwd19cadd1ad6qw26gr7h7hkh7m-fish-lsp-unstable-2024-07-26/share/fish/vendor_completions.d/fish-lsp.fish' does not exist or has zero size
```

Reply:
I think we should add those packages to the wrapper if they are sometimes used at runtime.

Like this patch:

```diff
diff --git a/pkgs/by-name/fi/fish-lsp/package.nix b/pkgs/by-name/fi/fish-lsp/package.nix
index bc9ceb51ecb8..cefbdbaf366a 100644
--- a/pkgs/by-name/fi/fish-lsp/package.nix
+++ b/pkgs/by-name/fi/fish-lsp/package.nix
@@ -36,9 +36,7 @@ stdenv.mkDerivation rec {
     nodejs
     installShellFiles
     makeWrapper
-    # fish-lsp dependencies
     fish
-    which
   ];
 
   yarnBuildScript = "setup";
@@ -54,7 +52,8 @@ stdenv.mkDerivation rec {
     cp -r . $out/share/fish-lsp
 
     makeWrapper ${lib.getExe nodejs} "$out/bin/fish-lsp" \
-      --add-flags "$out/share/fish-lsp/out/cli.js"
+      --add-flags "$out/share/fish-lsp/out/cli.js" \
+      --prefix PATH : "${lib.makeBinPath [fish which]}"
 
     ${lib.optionalString stdenv.buildPlatform.canExecute stdenv.hostPlatform ''
       installShellCompletion --cmd fish-lsp \
```

Reply:
Good idea, fixed